tests/jenkins/TestDockerBuild.groovy

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import jenkins.tests.BuildPipelineTest
+import org.junit.Before
+import org.junit.Test
+import static com.lesfurets.jenkins.unit.MethodCall.callArgsToString
+import static org.assertj.core.api.Assertions.assertThat
+
+class TestDockerBuild extends BuildPipelineTest {
+
+    // Variables
+    String dockerBuildGitRespository = 'https://github.com/opensearch-project/opensearch-build'
+    String dockerBuildGitRespositoryReference = 'main'
+    String dockerBuildScriptwithCommands = 'bash docker/ci/build-image-multi-arch.sh -v <TAG_NAME> -f <DOCKERFILE PATH>'
+
+    @Override
+    @Before
+    void setUp() {
+
+        super.setUp()
+
+        binding.setVariable('DOCKER_BUILD_GIT_REPOSITORY', dockerBuildGitRespository)
+        binding.setVariable('DOCKER_BUILD_GIT_REPOSITORY_REFERENCE', dockerBuildGitRespositoryReference)
+        binding.setVariable('DOCKER_BUILD_SCRIPT_WITH_COMMANDS', dockerBuildScriptwithCommands)
+
+    }
+
+    @Test
+    void DockerBuildRegression() {
+        super.testPipeline('jenkins/docker/docker-build.jenkinsfile',
+                'tests/jenkins/jenkinsjob-regression-files/docker/docker-build.jenkinsfile')
+    }
+
+    @Test
+    public void DockerBuildExecuteWithoutErrors() {
+        runScript("jenkins/docker/docker-build.jenkinsfile")
+
+        assertJobStatusSuccess()
+
+        // Ensure 'docker login' is executed in an external shell script
+        assertThat(helper.callStack.findAll { call ->
+            call.methodName == 'sh'
+        }.any { call ->
+            callArgsToString(call).contains('docker login')
```

Comment:
Can you add the entire line here instead of just ` docker login`?
```
docker logout && echo $DOCKER_PASSWORD | docker login -u $DOCKER_USERNAME --password-stdin && eval $DOCKER_BUILD_SCRIPT_WITH_COMMANDS
```

Reply:
thanks for the feedback, @gaiksaya . I got the test cases updated.. Please have another review and comments.